pkg/devfile/parser/data/1.0.0/components.go

```diff
@@ -24,17 +40,157 @@ func (d *Devfile100) GetAliasedComponents() []common.DevfileComponent {
 
 // GetProjects returns the slice of DevfileProject objects parsed from the Devfile
 func (d *Devfile100) GetProjects() []common.DevfileProject {
-	return d.Projects
+
+	var projects []common.DevfileProject
+	for _, v := range d.Projects {
+		// We are only supporting ProjectType git in V1
+		if v.Source.Type == ProjectTypeGit {
+			projects = append(projects, convertV1ProjectToCommon(v))
+		}
+	}
+
+	return projects
 }
 
 // GetCommands returns the slice of DevfileCommand objects parsed from the Devfile
 func (d *Devfile100) GetCommands() []common.DevfileCommand {
+
 	var commands []common.DevfileCommand
+	for _, v := range d.Commands {
+		cmd := convertV1CommandToCommon(v)
 
-	for _, command := range d.Commands {
-		command.Name = strings.ToLower(command.Name)
-		commands = append(commands, command)
+		commands = append(commands, cmd)
 	}
 
 	return commands
 }
+
+func (d *Devfile100) GetParent() common.DevfileParent {
+	return common.DevfileParent{}
+
+}
+
+func (d *Devfile100) GetEvents() common.DevfileEvents {
+	return common.DevfileEvents{}
+
+}
+
+func convertV1CommandToCommon(c Command) (d common.DevfileCommand) {
+	var exec common.Exec
+
+	for _, action := range c.Actions {
+
+		if *action.Type == DevfileCommandTypeExec {
+			exec = common.Exec{
+				Attributes:  c.Attributes,
+				CommandLine: *action.Command,
+				Component:   *action.Component,
+				Group:       getGroup(c.Name),
+				Id:          c.Name,
+				WorkingDir:  action.Workdir,
+				// Env:
+				// Label:
+			}
+		}
+
+	}
+
+	// TODO: Previewurl
+	return common.DevfileCommand{
+		//TODO(adi): Type
+		Exec: &exec,
+	}
+}
+
+func convertV1ComponentToCommon(c Component) (d common.DevfileComponent) {
+
+	var endpoints []*common.Endpoint
+	for _, v := range c.ComponentDockerimage.Endpoints {
+		endpoints = append(endpoints, convertV1EndpointsToCommon(v))
+	}
+
+	var envs []*common.Env
+	for _, v := range c.ComponentDockerimage.Env {
+		envs = append(envs, convertV1EnvToCommon(v))
+	}
+
+	var volumes []*common.VolumeMount
+	for _, v := range c.ComponentDockerimage.Volumes {
+		volumes = append(volumes, convertV1VolumeToCommon(v))
+	}
+
+	container := common.Container{
+		Name:         *c.Alias,
+		Endpoints:    endpoints,
+		Env:          envs,
+		Image:        *c.ComponentDockerimage.Image,
```

Comment:
null checks needed here?

Reply:
you are reviewing an previous commit, `string pointers` has been removed at many places.